WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/MediaUploadCompletionProcessor.java

```diff
@@ -40,23 +40,30 @@ public String processContent(String content) {
         if (headerMatcher.find()) {
             positionBlockStart = headerMatcher.start();
             String blockType = headerMatcher.group(1);
-            Matcher blockBoundaryMatcher = Pattern.compile(String.format(PATTERN_TEMPLATE_BLOCK_BOUNDARY, blockType),
-                    Pattern.DOTALL).matcher(content.substring(headerMatcher.end()));
+            String blockTagSuffix = headerMatcher.group(2);
+            Boolean isSelfClosingTag = blockTagSuffix.equals("/-->");
+            if (isSelfClosingTag) {
+                positionBlockEnd = headerMatcher.end();
```

Comment:
When the block uses the self-closing tag, we don't need to loop over the content to find the end position.